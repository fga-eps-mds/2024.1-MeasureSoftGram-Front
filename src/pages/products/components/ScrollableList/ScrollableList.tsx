import React, { useRef, useState } from 'react';
import { List, ListItem, IconButton, Box } from '@mui/material';
import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';
import { Organization } from '@customTypes/organization';


interface ScrollableListProps {
  organizationList: Organization[];
  onSelect: (organization: Organization) => void;
}

const ScrollableList: React.FC<ScrollableListProps> = ({ organizationList, onSelect }) => {
  const listRef = useRef<HTMLDivElement>(null);
  const [selectedIndex, setSelectedIndex] = useState<number | null>(null);

  const handleScrollUp = () => {
    if (listRef.current) {
      listRef.current.scrollBy({ top: -50, behavior: 'smooth' });
    }
  };

  const handleScrollDown = () => {
    if (listRef.current) {
      listRef.current.scrollBy({ top: 50, behavior: 'smooth' });
    }
  };

  const handleItemClick = (index: number) => {
    setSelectedIndex(index);
    onSelect(organizationList[index]);
  };

  console.log({ organizationList })

  return (
    <Box sx={{ width: 200, maxHeight: 350, overflow: 'hidden', position: 'relative' }}>
      <Box sx={{ display: 'flex', justifyContent: 'center' }}>
        <IconButton onClick={handleScrollUp}>
          <ArrowUpwardIcon />
        </IconButton>
      </Box>

      <Box ref={listRef} sx={{ maxHeight: 250, overflowY: 'auto', mt: 1, mb: 1 }}>
        <List>
          {organizationList.map((organization, index) => (
            <ListItem
              key={organization.id}
              onClick={() => handleItemClick(index)}
              sx={{
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                cursor: 'pointer',
                color: selectedIndex === index ? '#2B4D6F' : 'inherit',
                '&:hover': { backgroundColor: '#F4F5F6' }
              }}
            >
              {organization.name}
            </ListItem>
          ))}
        </List>
      </Box>

      <Box sx={{ display: 'flex', justifyContent: 'center' }}>
        <IconButton onClick={handleScrollDown}>
          <ArrowDownwardIcon />
        </IconButton>
      </Box>
    </Box>
  );
};

export default ScrollableList;
