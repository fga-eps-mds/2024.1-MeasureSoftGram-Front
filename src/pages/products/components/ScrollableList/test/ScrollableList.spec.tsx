import React from 'react';
import { render } from '@testing-library/react';
import ScrollableList from '../ScrollableList';

describe('ScrollableList Component', () => {

  it('Deve corresponder ao Snapshot', () => {
    const tree = render(<ScrollableList organizationList={[{
      id: "1", name: 'Organization 1', 'description': "string", 'url': 'abc',
      key: '',
      products: [],
    }]}
      onSelect={() => { }}
    />);
    expect(tree).toMatchSnapshot();
  });
});
