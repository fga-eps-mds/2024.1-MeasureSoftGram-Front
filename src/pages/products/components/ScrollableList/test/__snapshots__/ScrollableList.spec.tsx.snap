// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollableList Component Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-dmmwmi"
      >
        <div
          class="MuiBox-root css-1aimuqw"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowUpwardIcon"
              focusable="false"
              style="font-size: 1.8em;"
              viewBox="0 0 24 24"
            >
              <path
                d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-mhxp1c"
        >
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-c4b4dx-MuiListItem-root"
            >
              Organization 1
            </li>
          </ul>
        </div>
        <div
          class="MuiBox-root css-1aimuqw"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDownwardIcon"
              focusable="false"
              style="font-size: 1.8em;"
              viewBox="0 0 24 24"
            >
              <path
                d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-dmmwmi"
    >
      <div
        class="MuiBox-root css-1aimuqw"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowUpwardIcon"
            focusable="false"
            style="font-size: 1.8em;"
            viewBox="0 0 24 24"
          >
            <path
              d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-mhxp1c"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-c4b4dx-MuiListItem-root"
          >
            Organization 1
          </li>
        </ul>
      </div>
      <div
        class="MuiBox-root css-1aimuqw"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowDownwardIcon"
            focusable="false"
            style="font-size: 1.8em;"
            viewBox="0 0 24 24"
          >
            <path
              d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
